test/testframe.c

```diff
@@ -297,33 +346,180 @@ void
+                    MESSAGE(2, ("HDF5 was not built with multi-threaded support; Skipping test\n"));
+                    TestAlarmOff();
+                    continue;
+                }      
+#else
+                pthread_t *threads;
+                TestThreadArgs *thread_args;
+                int ret = 0;
+
+                if (max_num_threads <= 0) {
+                    fprintf(stderr, "Invalid number of threads specified\n");
+                    exit(EXIT_FAILURE);
+                }
+
+                threads = (pthread_t *)calloc((size_t) max_num_threads, sizeof(pthread_t));
+                thread_args = (TestThreadArgs *)calloc((size_t) max_num_threads, sizeof(TestThreadArgs));
+
+                if (!mt_initialized) {
+                    if (H5_mt_test_global_setup() < 0) {
+                        fprintf(stderr, "Error setting up global MT test info\n");
+                        exit(EXIT_FAILURE);
+                    }
+
+                    mt_initialized = true;
+                }
+
+                for (int i = 0; i < max_num_threads; i++) {
+                        thread_args[i].ThreadIndex = i;
+                        thread_args[i].Call = Test[Loop].Call;
+
+                        ret = pthread_create(&threads[i], NULL, PerformThreadTest, (void*) &thread_args[i]);
+
+                        if (ret != 0) {
+                            fprintf(stderr, "Error creating thread %d\n", i);
+                            exit(EXIT_FAILURE);
+                        }
+                }
+
+                for (int i = 0; i < max_num_threads; i++) {
+                        ret = pthread_join(threads[i], NULL);
+
+                        if (ret != 0) {
+                            fprintf(stderr, "Error joining thread %d\n", i);
+                            exit(EXIT_FAILURE);
+                        }
+                    }
+                
+
+                free(threads);
+                free(thread_args);
+                
+                TestAlarmOff();
+
+                test_num_errs = H5_ATOMIC_LOAD(Test[Loop].NumErrors);
+                H5_ATOMIC_STORE(Test[Loop].NumErrors, num_errs_g - test_num_errs);
+                MESSAGE(5, ("===============================================\n"));
+                MESSAGE(5, ("There were %d errors detected.\n\n", (int)H5_ATOMIC_LOAD(Test[Loop].NumErrors)));
+#endif /* H5_HAVE_MULTITHREAD */
+            }
         }
+    }
 
     Test_parameters = NULL; /* clear it. */
 
-    MESSAGE(2, ("\n\n"));
-    if (num_errs)
-        MESSAGE(VERBO_NONE, ("!!! %d Error(s) were detected !!!\n\n", (int)num_errs));
+    if (num_errs_g)
+        print_func("!!! %d Error(s) were detected !!!\n\n", (int)num_errs_g);
     else
         MESSAGE(VERBO_NONE, ("All tests were successful. \n\n"));
 }
 
+#ifdef H5_HAVE_MULTITHREAD
+/*
+ * Set up and execute a test flagged for multi-threaded
+ *   execution within a single thread.
+ */
+void *PerformThreadTest(void *test)
```

Comment:
Is this function used anywhere currently? I don't recall seeing it used, but might have missed a usage.